### PR TITLE
Replace Particle.dump_table with Particle.to_dict to remove strong coupling in that method

### PR DESCRIPTION
**Pull request by eduardo-rodrigues**
Addresses https://github.com/scikit-hep/particle/issues/233.

### CODE REVIEW

**Comment by eduardo-rodrigues** [comment]
I will merge this tomorrow unless you have any feedback.

File: src/particle/particle/convert.py [hunk 1]
```diff
@@ -109,6 +109,9 @@ def filter_file(fileobject):
         if not line.lstrip("\ufeff").lstrip().startswith("*"):
             stream.write(line)
     stream.seek(0)
+
+    fileobject.close()
```
Comment (henryiii) [inline src/particle/particle/convert.py:113]: Shouldn't you only close it if you opened it?

File: src/particle/particle/particle.py [hunk 2]
```diff
@@ -316,53 +312,68 @@ def dump_table(
...
+
+        >>> query_as_list = Particle.to_list(exclusive_fields=['pdgid', 'pdg_name'], n_rows=5)
+        >>> from tabulate import tabulate    # doctest: +SKIP
+        >>> print(tabulate(query_as_list, headers='firstrow'))    # doctest: +SKIP
+
+        Request the properties of a specific list of particles:
+
+        >>> query_as_list = Particle.to_list(filter_fn=lambda p: p.pdgid.is_lepton and p.charge!=0, exclusive_fields=['pdgid', 'name', 'mass', 'charge'])
+
+        >>> print(tabulate(query_as_list, headers='firstrow', tablefmt="rst", floatfmt=".12g", numalign="decimal"))    # doctest: +SKIP
```
Comment (henryiii) [inline src/particle/particle/particle.py:358]: You can add tabulate as a doc requirement, then not skip this.

File: src/particle/particle/convert.py (hunk 1 above)
Comment (eduardo-rodrigues) [reply src/particle/particle/convert.py:113]: Absolutely. It is a possibility to call the method with a file opened. I will add a check.

File: src/particle/particle/particle.py (hunk 2 above)
Comment (eduardo-rodrigues) [reply src/particle/particle/particle.py]: OK, done.

**Pull request merged**